src/main/scala/backends/futil/FutilBackend.scala

```diff
@@ -7,6 +7,17 @@ import Syntax._
 import Configuration._
 import CompilerError._
 
+/** Helper class that gives names to the fields of the output of `emitExpr` and `emitBinop`.
+  *  - `port` holds either an input or output port that represents how data flows between exprs
+  *  - `done` holds the port that signals when the writing or reading from `port` is done
+  *  - `structure` represents additional structure involved in computing the expression
+  */
+private class EmitOutput(
```

Comment:
Probably want to make this a case class so you don't need to call `new` down below.